src/Core/Maintenance/Staging/Handler/StagingExtensionHandler.php

```diff
@@ -0,0 +1,80 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Maintenance\Staging\Handler;
+
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsAnyFilter;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Plugin\KernelPluginLoader\ComposerPluginLoader;
+use Shopware\Core\Framework\Store\Services\AbstractExtensionDataProvider;
+use Shopware\Core\Framework\Store\Services\AbstractExtensionLifecycle;
+use Shopware\Core\Kernel;
+use Shopware\Core\Maintenance\Staging\Event\SetupStagingEvent;
+
+/**
+ * @internal
+ */
+#[Package('framework')]
+readonly class StagingExtensionHandler
+{
+    /**
+     * @param list<string> $extensionsToDisable
+     */
+    public function __construct(
+        private Kernel $kernel,
+        private AbstractExtensionDataProvider $extensionDataProvider,
+        private AbstractExtensionLifecycle $extensionLifecycleService,
+        private array $extensionsToDisable = [],
```

Comment:
i recently refactored the event to include the config data, that makes it easier in cases where you can not depend on the config values (think multi tenant system ;) ) -> would be nice if we could refactor this one as well

so the command get's the config value injected, put's that in the event and the handler here reads the data from the event

Additional benefit -> you could manipulate the config via the event as well if you want